### PR TITLE
feat/adding avaiable flag to the product datalayer

### PR DESCRIPTION
**Pull request by cesarocampov**
#### What is the purpose of this pull request?

 Adding a flag to the Product Data Layer to identify if the product is available or not. 

#### What problem is this solving?

Today the Product Data Layer Object does not differentiate between available and unavailable products. 
With this flag we are allowing customers to Filter this on their Google analytics.
The main use case is when working with pageViews on Google Analytics, customers today don't have a visibility of what products are available and not. offering this capability will allow them to focalice efforts to the products that are not available but receiving tons of visits. 

#### How should this be manually tested?
This Product is unavailable: https://tagmanager--arcaplanet.myvtex.com/collare-rogz-nylon-snake-mis-m-rossa-mm-16/p
1.- right click and inspect 
2.- go to console and type window.dataLayer
3.- open the productDetail event and check the flag isAvailable=false

This product is available: https://tagmanager--arcaplanet.myvtex.com/seresto-collare-antiparassitario-cani-oltre-8-kg/p
1.- right click and inspect 
2.- go to console and type window.dataLayer
3.- open the productDetail event and check the flag isAvailable=true

* [ ] New feature (a non-breaking change which adds functionality)

### CODE REVIEW

**Comment by victorhmp** [comment]
Hi @cesarocampov !
The tests are failing on this branch, could you check that?

**Comment by cesarocampov** [comment]
> Hi @cesarocampov ! The tests are failing on this branch, could you check that?

Hi @victorhmp the tests are passing now, thanks for the notice

File: react/modules/enhancedEcommerceEvents.ts [hunk 1]
```diff
@@ -71,6 +75,7 @@ export async function sendEnhancedEcommerceEvents(e: PixelMessage) {
                 dimension2: skuReferenceId ?? '',
                 dimension3: selectedSku.name,
                 price,
+                isAvailable,
```
Comment (icazevedo) [inline react/modules/enhancedEcommerceEvents.ts:78]: I don't think it should be called isAvailable.
```suggestion
                dimension4: isAvailable,
```

File: react/__tests__/index.test.tsx [hunk 2]
```diff
@@ -77,6 +77,7 @@ test('productDetail', () => {
             dimension1: '',
             dimension2: '12531',
             dimension3: 'Classic Pink',
+            isAvailable: true,
```
Comment (icazevedo) [inline react/__tests__/index.test.tsx:80]:
```suggestion
            dimension4: true,
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added 
+-`isAvailable` flag to the productDataLayer
```
Comment (icazevedo) [inline CHANGELOG.md:10]:
```suggestion
-`dimension4` property to the product data layer on `productView` events, representing the availability of the product
```

**Review by icazevedo — CHANGES_REQUESTED** [review]

File: react/modules/enhancedEcommerceEvents.ts (hunk 1 above)
Comment (cesarocampov) [reply react/modules/enhancedEcommerceEvents.ts]: @icazevedo I changed the value of the property to isAvailable. Given that it was a boolean I also made the possible values available or unavailable, this way it is more descriptive for the customers what do you think?
https://tagmanager--arcaplanet.myvtex.com/monge-medium-adult-ricco-di-pollo/p -> available product
https://tagmanager--arcaplanet.myvtex.com/seresto-collare-antiparassitario-cani-oltre-8-kg/p -> unavailable product

File: react/__tests__/index.test.tsx (hunk 2 above)
Comment (cesarocampov) [reply react/__tests__/index.test.tsx]: @icazevedo done!

File: CHANGELOG.md (hunk 3 above)
Comment (cesarocampov) [reply CHANGELOG.md]: @icazevedo done!

**Review by icazevedo — APPROVED** [review]

**Review by igorbrasileiro — APPROVED** [review]

**Pull request merged**